$NetBSD: patch-af,v 1.2 2009/12/17 18:23:49 abs Exp $

Tk uses X event numbers to index this array, as well as adding a few of its
own, starting with "VirtualEvent" (Which is correctly set to LASTEvent,
defined in include/X11/X.h).  In xproto-7.0.13, a new event "GenericEvent" was
added - making the array broken for all of Tk's internal events.

A cleaner fix, but a much bigger patch, is listed in the Tk bug tracker:

http://sourceforge.net/tracker/index.php?func=detail&aid=2010422&group_id=12997&atid=112997

--- generic/tkBind.c.orig	2001-10-13 01:25:10.000000000 +0000
+++ generic/tkBind.c
@@ -575,6 +575,9 @@ static int flagArray[TK_LASTEVENT] = {
    /* ColormapNotify */		COLORMAP,
    /* ClientMessage */		0,
    /* MappingNotify */		0,
+#ifdef GenericEvent
+   /* GenericEvent */		0, /* Introduced in xproto-7.0.13 */
+#endif
    /* VirtualEvent */		VIRTUAL,
    /* Activate */		ACTIVATE,	    
    /* Deactivate */		ACTIVATE,
